scripts/lib/devtool/ide.py

```diff
@@ -1030,6 +1049,9 @@ def vscode_tasks(self, args, gdb_cross):
                     "label": pretty_id,
                     "type": "shell",
                     "isBackground": True,
+                    "dependsOn": [
+                        "install && deploy-target %s" % self.bpn
```

Comment:
Yes that is closer to the traditional VSCode experience where changes in the code should be always reflected in the debugging session (the extension for pytest or nodejs do it for instance).

With my change however it is not possible to debug without deploying. But that would be risky since the executed code would not correspond to the current code if it was modified. Lines would not match for example. Please tell me if this breaks a specific use case for your original decision that we could consider supporting in other ways.

Reply:
The ideal solution would be to always call the deploy script (what your commit does) but also improve the generated deploy script to skip the deployment if nothing changed since the last deployment.

One idea would be to make the script which is generated by the function `gen_fakeroot_install_script` writing a timestamp to a file. (Something like ``cmd_lines.append('date +%s > last_updated')``)

Also, the Python script generated by "gen_deploy_target_script" could write a timestamp to a file when it deploys the files to the target device. Finally, the Python script can read the timestamps from both files and compare them. If the last deployment occurred after the last update, the deployment script can simply print "deployment skipped" and exit without redeploying. 

This would not only speed up the workflow significantly. It would also allow that between two debugging sessions e.g. a config file on the device is changed without it being automatically overwritten again.

Reply:
maybe the  ``install && deploy-target %s" % self.bpn`` and deploy should be a variable. The task name and the dependency name must be equal.

Reply:
> The ideal solution would be to always call the deploy script (what your commit does) but also improve the generated deploy script to skip the deployment if nothing changed since the last deployment.
> 
> One idea would be to make the script which is generated by the function `gen_fakeroot_install_script` removing a stamp file and the Python script generated by "gen_deploy_target_script" writing a stamp file and only deploy if the stamp file does not exist.
> 
> This would not only speed up the workflow significantly. It would also allow that between two debugging sessions e.g. a config file on the device is changed without it being automatically overwritten again.

it could speed things up for recipes producing huge binaries which are slow to install. I think this is rather a "nice to have" at this point.

Reply:
> install && deploy-target %s" % self.bpn

Yes, it's a slight code duplication. I've updated the commit with a variable.